script.js

```diff
@@ -0,0 +1,165 @@
+const toDoForm = document.querySelector("form");
+const toDoInput = toDoForm.querySelector(".input-text");
+const toDoButton = toDoForm.querySelector(".input-button");
+
+const TODO_DB = "todo";
+let toDoList = [];
+
+// localStorage 모델: 웹이 로드 될 때 기존 투두리스트를 가져오기 위함
+const localStorageModel = {
+    // localStorage에 저장
+    save: () => {
+        localStorage.setItem(TODO_DB, JSON.stringify(toDoList));
+    },
+
+    // localStorage에서 로드
+    load: () => {
+        const loaded = localStorage.getItem(TODO_DB);
+        if (loaded !== null) {
+            const parsed = JSON.parse(loaded);
+            toDoList = parsed;
+        }
+    }
+}
+
+// List 모델
+const list = {
+    // 개수 계산
+    getCount: () => {
+        let pending = 0;
+        toDoList.forEach((item) => {
+            if (!item.done) {
+                pending++;
+            }
+        })
+        return {
+            pending: pending,
+            done: toDoList.length - pending
+        }
+    },
```

Comment:
하나의 함수는 하나의 일만을 시키는 것이 바람직합니다.
여기서 `getCount`라는 함수의 이름이 직관적이지 않을 뿐더러, return 하는 타입, return 하는 객체의 구조와 속성명들도 모호합니다.

다음 중 하나의 방법으로 고쳐볼 수 있겠네요.
1. `pending`, `done`를 `pendingTodosCount`, `doneTodosCount`로 변경하기
2. `getCount()` 함수 대신, `getPendingTodosCount()`, `getDoneTodosCount()`로 나누어 놓는 것이 좋을 것 같습니다. (꼭 이 이름일 필요는 없습니다. 자세하고, 명확하고, 일관성 있는 이름일수록 좋습니다)
- `countTodos()`도 좋은 것 같지만, 이왕이면 함수가 무엇인가를 반환한다는 것을 직접적으로 변수 이름에 드러내는 것(`get`)을 저는 개인적으로 선호합니다. 이 함수가 세기만 하는 게 아니라, 세고 반환을 하니까요?
```suggestion
    getPendingTodosCount() {
        return toDoList.filter(todoItem => !item.done).length
        // 이렇게 filter를 쓰면, 훨씬 더 쉽게 특정 조건을 만족하는 element의 리스트를 찾을 수 있습니다. 
    },
    getDoneTodosCount() {
        return toDoList.filter(todoItem => item.done).length
    }
```

Reply:
피드백 감사합니다. 저도 이 부분이 정말 마음에 걸렸거든요 ㅎㅎ... filter라는 메소드를 깜박헀습니다.
또한 네이밍에 대해서 짚어주셔서 감사합니다. 함수명을 지을 때 `동사 + 목적어 + 부사 + ...` 너무 길어지는 현상때문에 일부로 줄이는 습관이 들었었는데 고쳐야겠네요 😂